beat-that-script.js

```diff
@@ -0,0 +1,149 @@
+//REQUIREMENTS//
+//1) There are two players and players take turns
+//2) when a player clicks submit, the game rolls 2 dice and shows the dice rolls
+//3) the player picks the order of the dice they want.
+//4) after both players have rolled and chosen the dice order, the game compares both players' scores. Player with the higher combined number wins
+
+// set global variables
+var rollDiceState = "rollDiceState";
+var chooseDiceState = "chooseDiceState";
+var compareScoresState = "compareScoresState";
+var gameState = rollDiceState;
+
+var currentPlayerRolls = [];
+var currentPlayer = 1;
+var bothPlayersScore = [];
+
+// reset game automatically
+var resetGame = function () {
+  currentPlayer = 1;
+  gameState = rollDiceState;
+  bothPlayersScore = [];
+};
+
+//Dice roll function
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+
+  return randomInteger;
+};
+
+//Helper function to store both dice numbers in an array
+var rollBothDice = function () {
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+  return (
+    `Welcome, Player ` +
+    currentPlayer +
+    `.` +
+    `<br><br>You rolled: ${currentPlayerRolls[0]} for Dice One and ${currentPlayerRolls[1]} for Dice Two. <br><br> Next, please input '1' if you would like to choose Dice One and '2' if you would like to choose Dice Two as the first digit of your final value.`
+  );
+};
+
+//Helper function to get player's score
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  //input validation
+  if (playerInput != 1 && playerInput != 2) {
+    console.log(`Control flow: input !== 1 && input !== 2`);
+    return `Error! Please input:<br>'1' if you want to use Dice One as the first digit of your final value or<br> '2' if you want to use Dice Two as the first digit of your final value.<br><br>You rolled: <br>${currentPlayerRolls[0]} for Dice One and <br>
```

Comment:
Good use of input validation to ensure that users only input what we want them to